File: option/src/app/node/src/rest.js
const express = require('express')
const app = express()
const port = 8080

const oracledb = require('oracledb');

app.get('/info', (req, res) => {
    res.send('NodeJS - Express / Oracle')
})

app.get('/dept', async (req, res) => {
    let connection;

    try {
        connection = await oracledb.getConnection({ user: process.env.DB_USER, password: process.env.DB_PASSWORD, connectionString: process.env.DB_URL });
        result = await connection.execute(
            `select deptno, dname, loc from DEPT`,
            [],
            { resultSet: true, outFormat: oracledb.OUT_FORMAT_OBJECT });
        const rs = result.resultSet;
        let row;
        let arr = [];
        while ((row = await rs.getRow())) {
            arr.push(row);
        }
        await rs.close();
        res.send(arr)
    } catch (err) {
        console.error(err);
    } finally {
        if (connection) {
            try {
                await connection.close();
            } catch (err) {
                console.error(err);
            }
        }
    }
})

app.listen(port, () => {
    console.log(`OCI Starter: listening on port ${port}`)
})

File: option/src/app/python/src/rest.py
import os
import json
from app import app
from flask import jsonify
from flask import flash, request
import oracledb

@app.route('/dept')
def dept():
    try:
        conn = oracledb.connect(
          user=os.getenv('DB_USER'),
          password=os.getenv('DB_PASSWORD'),
          dsn=os.getenv('DB_URL'))
        print("Successfully connected to Oracle Database")
        cursor = conn.cursor()
        a = []
        for row in cursor.execute('select deptno, dname, loc from dept'):
            a.append( {"deptno": row[0], "dname": row[1], "loc": row[2]} )
        print(a)
        response = jsonify(a)
        response.status_code = 200
        return response
    except Exception as e:
        print(e)
    finally:
        cursor.close() 
        conn.close()     

@app.route('/info')
def info():
        return "Python - Flask / Oracle"          

if __name__ == "__main__":
    app.run(host="0.0.0.0", port=8080)